lib/mix/lib/mix/compilers/test.ex

```diff
@@ -103,6 +111,26 @@ defmodule Mix.Compilers.Test do
     end
   end
 
+  defp do_require(true, test_files, parallel_require_callbacks, warnings_as_errors?) do
+    # we are running with --trace-require, so while we still use the parallel compiler
+    # for the parallel_require_callbacks and warnings_as_errors?, but actually only
+    # require one file at a time
+    failed =
+      for file <- test_files do
+        do_require(false, [file], parallel_require_callbacks, warnings_as_errors?)
```

Comment:
I think we can use Code.require_file instead... but I wonder if that largely changes the semantics of the code?

Reply:
I changed it to still use the ParallelCompiler because of the `parallel_require_callbacks` and the `warnings_as_errors?` option (https://github.com/elixir-lang/elixir/pull/13589/commits/134eace96f82b10e950946e69a0abb7659d1ab64), otherwise `--stale` or `--warnings-as-errors` won't work if combined with `--sort-first/last` or `--trace-require`.